src/sql/src/plan/statement/ddl.rs

```diff
@@ -3056,6 +3091,19 @@ fn kafka_sink_builder(
         compression_type,
         progress_group_id,
         transactional_id,
+        topic_options: KafkaSinkTopicOptions {
+            partition_count: topic_partition_count,
+            replication_factor: topic_replication_factor,
+            topic_config,
+        },
+        progress_topic_options: KafkaSinkTopicOptions {
+            // We only allow configuring the progress topic replication factor for now.
```

Comment:
I'd also say "For correctness, the partition count MUST be one and for performance the compaction policy MUST be enabled."